### PR TITLE
Author Synonyms

### PR DESCRIPTION
**Pull request by urvishp80**
- update author synonyms with default author name by fetching from [authors-synonyms.csv](https://github.com/bitcoinsearch/synonyms/blob/main/authors-synonyms.csv)
- daily GitHub Actions file

### CODE REVIEW

File: update_authors_synonyms.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                name = literal_eval(name.strip())
+            except:
+                name = name.strip()
+            mapping[name] = uniform_name
+    if mapping:
+        logger.info(f"Authors synonyms fetched successfully!")
+    return mapping
+
+
+def remove_timestamps_from_author_names(author_list):
```
Comment (kouloumos) [inline update_authors_synonyms.py]: Why is this needed? Do we have occurrences where we scrape author names and they have timestamps in their name? If that's the case, can we do this cleaning when we are first scrapping this data?
In any case, a comment in the code here explaining the reasoning could be helpful for future reference.

**Review by kouloumos — COMMENTED** [review]
This looks good, but I have a question regarding the implementation:

This code iterates over all the documents. My understanding was that the manual solution that we are currently using (through the console) is an elastic search query that searches specifically for X and replaces with Y.
For example, in the case of "Wladimir J. van der Laan" it will execute 2 "find-and-replace" actions:
- for "Wladimir"->"Wladimir J. van der Laan"
- for "wumpus"->"Wladimir J. van der Laan"

Am I right here? Is that what we are currently doing? This seems faster as it doesn't iterate over all the documents. Am I missing something here? Is the performance the same because of how elastic search indexes documents and searches for documents? Can you elaborate on your current solution vs what I am proposing?

File: update_authors_synonyms.py (hunk 1 above)
Comment (urvishp80) [reply update_authors_synonyms.py]: Yes, there were some occurrences of timestamps in some older files of previous bitcoin-dev (old mailing-list), let me update those docs locally as it is one time thing. Additionally I've update the existing script to handle this issue on the fly while scraping so It won't be required here. I'll make sure to add comments in the scripts.

**Comment by urvishp80** [comment]
Thank you for your suggestion @kouloumos, 'Find and Replace' approach would be the correct way to update the author synonyms. Let me quickly update the script.

File: common/elasticsearch_utils.py [hunk 2]
```diff
@@ -107,3 +109,94 @@ def upsert_document(index_name, doc_id, doc_body):
...
+
+def get_domain_query(url):
+    if isinstance(url, list):
+        domain_query = {"terms": {"domain.keyword": url}}
+    else:
+        domain_query = {"term": {"domain.keyword": url}}
+    return domain_query
+
+
+def extract_data_from_es(index, domain):
```
Comment (kouloumos) [inline common/elasticsearch_utils.py]: We are not using this method now. No need to be part of this PR

File: common/elasticsearch_utils.py [hunk 3]
```diff
@@ -107,3 +109,94 @@ def upsert_document(index_name, doc_id, doc_body):
     # Perform the upsert operation
     response = es.update(index=index_name, id=doc_id, body=request_body)
     return response
+
+
+def get_domain_query(url):
```
Comment (kouloumos) [inline common/elasticsearch_utils.py]: We are not using this method now. No need to be part of this PR

**Review by kouloumos — APPROVED** [review]
Looks good!

I left comments inline for 2 unused methods. I think it's better to remove them and bring them when/if we need them down the road.

**Pull request merged**